### PR TITLE
Porting aid

### PR DESCRIPTION
**Pull request by kojo1**
XFOPEN and other file access API test
WOLFSSL_PASSTHRU_ERR: passing through errors for acquire as much information as possible
check XFSEEK return value

### CODE REVIEW

File: src/ssl.c [hunk 1]
```diff
@@ -6283,7 +6285,8 @@ static int wolfSSL_SetTmpDH_file_wrapper(WOLFSSL_CTX* ctx, WOLFSSL* ssl,
 
     file = XFOPEN(fname, "rb");
     if (file == XBADFILE) return WOLFSSL_BAD_FILE;
-    XFSEEK(file, 0, XSEEK_END);
+    if(XFSEEK(file, 0, XSEEK_END) != 0)
+        return WOLFSSL_BAD_FILE;;
```
Comment (cconlon) [inline src/ssl.c]: Please remove double semicolon here (;;).

File: src/ssl.c [hunk 2]
```diff
@@ -21510,7 +21521,8 @@ int wolfSSL_cmp_peer_cert_to_file(WOLFSSL* ssl, const char *fname)
         if (file == XBADFILE)
             return WOLFSSL_BAD_FILE;
 
-        XFSEEK(file, 0, XSEEK_END);
+        if(XFSEEK(file, 0, XSEEK_END) != 0)
+            return WOLFSSL_BAD_FILE;;
```
Comment (cconlon) [inline src/ssl.c]: Please remove double semicolon here (;;).

File: src/ssl.c (hunk 1 above)
Comment (kojo1) [reply src/ssl.c]: Done

File: src/ssl.c (hunk 2 above)
Comment (kojo1) [reply src/ssl.c]: Done

**Pull request merged**